### PR TITLE
[W3.10][W09-A1]Yu Hong Herald

### PR DESCRIPTION
**Pull request by yuhongherald**
Ready for review.

### CODE REVIEW

**Comment by yuhongherald** [comment]
@eugenepeh just briefly will do (5.11 is similar)

File: src/seedu/addressbook/AddressBook.java [hunk 1]
```diff
@@ -579,6 +588,47 @@ private static String executeListAllPersonsInAddressBook() {
         return getMessageForPersonsDisplayedSummary(toBeDisplayed);
     }
 
+    /**
+     * Displays all persons in the address book to the user; sorted by second command word
+     * @return feedback display message for the operation result
+     */
+    private static String executeSortAllPersonsInAddressBook(String arg) {
```
Comment (eugenepeh) [inline src/seedu/addressbook/AddressBook.java:595]: Normally function name can be shorten to highlight the key important points e.g. `executeSortAddressBook` or `executeSortAllPersons` both sort of implies the other.

File: src/seedu/addressbook/AddressBook.java [hunk 2]
```diff
@@ -579,6 +588,47 @@ private static String executeListAllPersonsInAddressBook() {
...
+        ArrayList<String[]> toBeSorted = new ArrayList<>(getAllPersonsInAddressBook());
+        ArrayList<String[]> toBeDisplayed = new ArrayList<>();
+        String[] nextPerson;
+
+        while (!toBeSorted.isEmpty())
+        {
+            nextPerson = toBeSorted.get(0);
+            for (String[] person: toBeSorted)
+            {
+                if (isSecondPersonSmaller(nextPerson, person, commandType)) {
```
Comment (eugenepeh) [inline src/seedu/addressbook/AddressBook.java:615]: Here seems like an arrow head :)

File: src/seedu/addressbook/AddressBook.java [hunk 3]
```diff
@@ -579,6 +588,47 @@ private static String executeListAllPersonsInAddressBook() {
...
+                    nextPerson = person;
+                }
+            }
+            toBeDisplayed.add(nextPerson);
+            toBeSorted.remove((nextPerson));
+        }
+        showToUser(toBeDisplayed);
+        return getMessageForPersonsDisplayedSummary(toBeDisplayed);
+
+    }
```
Comment (eugenepeh) [inline src/seedu/addressbook/AddressBook.java:625]: this entire method seems a bit too long, perhaps can increase the SLAP (Single level abstraction principle)

File: src/seedu/addressbook/AddressBook.java [hunk 4]
```diff
@@ -129,6 +129,13 @@
     private static final String COMMAND_HELP_DESC = "Shows program usage instructions.";
     private static final String COMMAND_HELP_EXAMPLE = COMMAND_HELP_WORD;
 
+    private static final String COMMAND_SORT_WORD = "sort";
+    private static final String COMMAND_SORT_DESC = "Displays all persons as a list, sorted by specification. Does " +
+            "not save sorted list. Fields are name, phone or address";
```
Comment (eugenepeh) [inline src/seedu/addressbook/AddressBook.java:134]: `+` symbol should be on the next line according to our coding standard.

**Review by eugenepeh — COMMENTED** [review]
Good job on completing the activity.
Some comments added. Please close the PR after reading the comments.

**Pull request closed without merging**